### ISSUE TITLE
MSlice will not recognise a cut when the parent is not in the ADS

### ISSUE THREAD

**Issue by EltayebAhmed** [mantidproject/mslice#51]
A possible solution would be to reimplement CutAlgorith.is_cut and CutAlgorithm.get_cut_params without depending on the workspace history.
MSlice currently checks to see if a workspace is a cut by looping through the workspace history and searching for a call to BinMD.
MSlice extracts the parameters by looking at the properties passed to the BinMD algorithm.
This is superfluous and will fail if the parent algorithm does not exist in the ADS. 
To be exact 

```
        history = cut_workspace.getHistory()
        for i in range(history.size()-1,-1,-1):
            algorithm = history.getAlgorithm(i) #<- RuntimeError will be raised here 
```

The above snippet will fail if run on a workspace in ipython in mantid on workspace
 even though right-clicking the smae workspace and clicking show History will bring up the entire workspaces history

It is possible to deduct the cut axis specs by doing something along the lines of

```
dim = workspace.getDimension(0)
name = dim.getName()
min = dim.getMinimum()
max = dim.getMaximum()
step = dim.getX(1) - dim.getX(0)
```

All that is left is to find a way to find the integration limits without failing if parent has been removed from the ADS

**Comment by martyngigg** [comment]
@EltayebAhmed Please provide more details of what the actual issue is. Most of what we discussed on slack would be good to put in the description for reference. If you don't get to it we need to preserve the information.

**Comment by EltayebAhmed** [comment]
If you right-click on a workspace in mantidPlot and click show history it will  bring up the entire history regardless of if any of the parents are in the ADS or not

**Comment by MatthewBowles** [comment]
Fixed in #222
